pkg/controllers/rollout/controller.go

```diff
@@ -643,6 +663,72 @@ func (r *Reconciler) SetupWithManager(mgr runtime.Manager) error {
 		Complete(r)
 }
 
+// handleClusterResourceOverrideSnapshot parse the clusterResourceOverrideSnapshot label and enqueue the CRP name associated
+// with the clusterResourceOverrideSnapshot if set.
+func handleClusterResourceOverrideSnapshot(o client.Object, q workqueue.RateLimitingInterface) {
```

Comment:
is there a way to combine these two functions? We will have RP and CRP co exist all over the place soon.

Reply:
we could, but need to rely on the if else condition to handle some cases based on the type, including logging, which hurts our readability.

For the current case, prefer to do the copy & paste.

Reply:
is there a way to use generic templating?

Reply:
it may be workable, either using https://google.github.io/styleguide/go/decisions#generics or interface.